dist/src/main/java/io/camunda/webapps/controllers/ForwardErrorController.java

```diff
@@ -0,0 +1,125 @@
+  public void init() {
+    // list of endpoints for which 404 is returned
+    ignoredRequestsMatchers =
+        List.of(
+            new AntPathRequestMatcher(PUBLIC_API),
+            new AntPathRequestMatcher(API),
+            new AntPathRequestMatcher(NO_PERMISSION),
+            new AntPathRequestMatcher(SSO_CALLBACK_URI),
+            new AntPathRequestMatcher(IDENTITY_CALLBACK_URI),
+            // actuator endpoints
+            new AntPathRequestMatcher(
+                Optional.ofNullable(environment.getProperty("management.endpoints.web.base-path"))
+                    .map(s -> s.concat("/**"))
+                    .orElse(DEFAULT_MANAGEMENT_URI)));
+  }
+
+  @RequestMapping(value = {"/{regex:[\\w-]+}", "/**/{regex:[\\w-]+}"})
+  public String forward404(final HttpServletRequest request, final HttpServletResponse response) {
+    if (ignoredRequestsMatchers.stream()
+        .anyMatch(requestMatcher -> requestMatcher.matches(request))) {
+      response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+      return null;
+    }
+    final String requestedURL = getRequestedURL(request);
+    if (!requestedURL.startsWith("/tasklist") && !requestedURL.startsWith("/operate")) {
+      if (isStandaloneTasklist()) {
+        return "redirect:/tasklist" + requestedURL;
+      } else {
+        return "redirect:/operate" + requestedURL;
+      }
+    } else if (isLoginDelegated() && isNotLoggedIn()) {
+      return saveRequestAndRedirectToLogin(request, requestedURL);
+    } else {
+      if (isStandaloneTasklist()) {
+        return "forward:/tasklist";
+      } else {
+        return "forward:/operate";
+      }
+    }
+  }
+
+  private boolean isStandaloneTasklist() {
+    return environment.acceptsProfiles(TASKLIST.getId(), "!" + OPERATE.getId());
+  }
+
+  private String getRequestedURL(final HttpServletRequest request) {
+    final String requestedUri =
+        request.getRequestURI().substring(request.getContextPath().length());
+    final String queryString = request.getQueryString();
+    if (StringUtils.isEmpty(queryString)) {
+      return requestedUri;
+    } else {
+      return requestedUri + "?" + queryString;
+    }
+  }
+
+  private String saveRequestAndRedirectToLogin(
+      final HttpServletRequest request, final String requestedURL) {
+    LOGGER.warn(
+        "Requested path {}, but not authenticated. Redirect to  {} ", requestedURL, LOGIN_RESOURCE);
+    request.getSession(true).setAttribute(REQUESTED_URL, requestedURL);
+    return "forward:" + LOGIN_RESOURCE;
+  }
+
+  private boolean isNotLoggedIn() {
+    final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+    return (authentication instanceof AnonymousAuthenticationToken)
+        || !authentication.isAuthenticated();
+  }
+
+  private boolean isLoginDelegated() {
+    return Arrays.stream(environment.getActiveProfiles())
+        .anyMatch(Set.of(IDENTITY_AUTH.getId(), SSO_AUTH.getId())::contains);
+  }
```

Comment:
Again, it should be a configuration and not depending on the profile.

Reply:
`isLoginDelegated` is a legacy method, but we can sort it out

Reply:
Done! new property `camunda.webapps.loginDeletgated` is introduced and set by WebappsConfigurationInitializer.java